connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/ConnectorUtil.java

```diff
@@ -86,4 +88,65 @@ public static InboundConnectorConfiguration getRequiredInboundConnectorConfigura
                         "InboundConnectorExecutable %s is missing @InboundConnector annotation",
                         cls)));
   }
+
+  private static String toNormalizedConnectorName(final String connectorName) {
+    return connectorName.trim().replaceAll("[^a-zA-Z0-9_ ]", "").replaceAll(" ", "_").toUpperCase();
+  }
+
+  private static String toConnectorTypeEnvVariable(final String normalizedConnectorName) {
+    return "CONNECTOR_" + normalizedConnectorName + "_TYPE";
+  }
+
+  private static String toConnectorTimeoutEnvVariable(final String normalizedConnectorName) {
+    return "CONNECTOR_" + normalizedConnectorName + "_TIMEOUT";
+  }
+
+  public static String[] getInputVariables(Class<?> cls, OutboundConnector annotation) {
+    List<ReflectionUtil.MethodWithAnnotation<Operation>> operations =
+        ReflectionUtil.getMethodsAnnotatedWith(cls, Operation.class);
+    if (operations.isEmpty()) {
+      return annotation.inputVariables();
+    } else {
+      return getInputVariables(operations).toArray(new String[0]);
+    }
+  }
+
+  public static Set<String> getInputVariables(
+      List<ReflectionUtil.MethodWithAnnotation<Operation>> operations) {
+    Set<String> variables = new HashSet<String>();
+    operations.forEach(
+        method -> {
+          method
+              .parameters()
+              .forEach(
+                  parameter -> {
+                    if (parameter.getAnnotation(io.camunda.connector.api.annotation.Variable.class)
+                        != null) {
+                      String variableName =
+                          parameter
+                              .getAnnotation(io.camunda.connector.api.annotation.Variable.class)
+                              .value();
+                      if (variableName.isEmpty()) {
+                        // When the variable name is empty, we assume that the variables are mapped
+                        // from the root
+                        for (Field declaredField : parameter.getType().getDeclaredFields()) {
+                          declaredField.setAccessible(true);
+                          String fieldName = declaredField.getName();
+                          variables.add(fieldName);
+                        }
+                      } else {
+                        // If the variable name contains a dot, we take the part before the dot as
+                        // the variable name
+                        if (variableName.contains(".")) {
+                          String[] parts = variableName.split("\\.");
+                          variables.add(parts[0]);
```

Comment:
Can this be unexpected in certain cases? Like if someone tries to use `parent.child` to reflect nesting 🤔

Reply:
Great question. My assumption was that the engine resolves only the top level variable name when fetching variables for jobs. Do you maybe have a reference for the path syntax is being used when requesting job variables?

Reply:
Nesting doesnt seem to be supported so "x.y.z" is a valid variable name.

Reply:
Ok good, I agree with the approach not to implement this for now 👍 Thanks for checking